executor-service/src/main/java/executor/service/facade/parallel/ConfigurableThreadPool.java

```diff
@@ -0,0 +1,53 @@
+package executor.service.facade.parallel;
+
+import executor.service.model.ThreadPoolConfig;
+
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.logging.Logger;
+
+public class ConfigurableThreadPool extends ThreadPoolExecutor {
+
+    private final ThreadLocal<Long> startTime = new ThreadLocal<>();
+    private final Logger log = Logger.getLogger("ConfigurableThreadPool");
+    private final AtomicLong numTasks = new AtomicLong();
+    private final AtomicLong totalTime = new AtomicLong();
+
+    public ConfigurableThreadPool(ThreadPoolConfig threadPoolConfig) {
+        super(threadPoolConfig.getCorePoolSize(), threadPoolConfig.getCorePoolSize(),
+                threadPoolConfig.getKeepAliveTime(), TimeUnit.MILLISECONDS, new LinkedBlockingQueue<>());
+    }
+
+    @Override
+    protected void beforeExecute(Thread t, Runnable r) {
+        super.beforeExecute(t, r);
+        log.fine(String.format("Thread %s: start %s", t, r));
+        startTime.set(System.nanoTime());
+    }
+
+    @Override
+    protected void afterExecute(Runnable r, Throwable t) {
+        try {
+            long endTime = System.nanoTime();
+            long taskTime = endTime - startTime.get();
+            numTasks.incrementAndGet();
+            totalTime.addAndGet(taskTime);
+            log.fine(String.format("Thread %s: end %s, time=%dns",
+                    t, r, taskTime));
+        } finally {
+            super.afterExecute(r, t);
+        }
+    }
+
+    @Override
+    protected void terminated() {
+        try {
+            log.info(String.format("Terminated: avg time=%dns",
+                    totalTime.get() / numTasks.get()));
```

Comment:
If `numTasks.get()` is zero this will result in an `ArithmeticException`
 
The purpose of this method is not clear, you don't need 
to override this method if you don't add any additional behavior to it